pkg/diagnose/cni.go

```diff
@@ -107,44 +107,40 @@ func detectCalicoConfigMap(clientSet kubernetes.Interface) (bool, error) {
 	return false, nil
 }
 
-func checkCalicoIPPoolsIfCalicoCNI(info *cluster.Info, status reporter.Interface) bool {
+func checkCalicoIPPoolsIfCalicoCNI(info *cluster.Info, status reporter.Interface) error {
 	status.Start("Trying to detect the Calico ConfigMap")
 	defer status.End()
 
 	found, err := detectCalicoConfigMap(info.ClientProducer.ForKubernetes())
 	if err != nil {
-		status.Failure("Error trying to detect the Calico ConfigMap: %s", err)
-		return false
+		return status.Error(err, "Error trying to detect the Calico ConfigMap")
 	}
 
 	if !found {
-		return true
+		return nil
 	}
 
 	status.Start("Calico CNI detected, checking if the Submariner IPPool pre-requisites are configured")
 
 	gateways, err := info.GetGateways()
 	if err != nil {
-		status.Failure("Error retrieving Gateways: %v", err)
-		return false
+		return status.Error(err, "Error retrieving Gateways")
 	}
 
 	if len(gateways) == 0 {
 		status.Warning("There are no gateways detected on the cluster")
-		return false
+		return errors.New("no gateways on the cluster")
```

Comment:
It's a little odd to log a warning yet fail the command - perhaps we should use `return status.Error(err, "No gateways detected on the cluster"`).